examples/node/domstubs.js

```diff
@@ -91,39 +91,102 @@ DOMElement.prototype = {
     }
   },
 
-  toString: function DOMElement_toString() {
-    var buf = [];
-    buf.push('<' + this.nodeName);
-    if (this.nodeName === 'svg:svg') {
-      buf.push(' xmlns:xlink="http://www.w3.org/1999/xlink"' +
-               ' xmlns:svg="http://www.w3.org/2000/svg"');
-    }
-    for (var i in this.attributes) {
-      buf.push(' ' + i + '="' + xmlEncode(this.attributes[i]) + '"');
-    }
-
-    buf.push('>');
-
-    if (this.nodeName === 'svg:tspan' || this.nodeName === 'svg:style') {
-      buf.push(xmlEncode(this.textContent));
-    } else {
-      this.childNodes.forEach(function(childNode) {
-        buf.push(childNode.toString());
-      });
-    }
-    buf.push('</' + this.nodeName + '>');
-    return buf.join('');
-  },
-
   cloneNode: function DOMElement_cloneNode() {
     var newNode = new DOMElement(this.nodeName);
     newNode.childNodes = this.childNodes;
     newNode.attributes = this.attributes;
     newNode.textContent = this.textContent;
     return newNode;
   },
+
+  // This method is offered for convenience. It is recommended to directly use
+  // getSerializer because that allows you to process the chunks as they come
+  // instead of requiring the whole image to fit in memory.
+  toString: function DOMElement_toString() {
+    var buf = [];
+    var serializer = this.getSerializer();
+    var chunk;
+    while ((chunk = serializer.getNext()) !== null) {
```

Comment:
@doublex I updated this PR, and changed this line, as follows:

```diff
--- a/examples/node/pdf2svg.js
+++ b/examples/node/pdf2svg.js
@@ -56,7 +56,7 @@ util.inherits(ReadableSVGStream, stream.Readable);
 // Implements https://nodejs.org/api/stream.html#stream_readable_read_size_1
 ReadableSVGStream.prototype._read = function() {
   var chunk;
-  while ((chunk = this.serializer.getNext())) {
+  while ((chunk = this.serializer.getNext()) !== null) {
     if (!this.push(chunk)) {
       return;
     }
```

This is necessary, because otherwise the SVG file is truncated too early (e.g. when a SVG node has no text content).